packages/bibtex/init.lua

```diff
@@ -7,37 +7,95 @@ local epnf = require("epnf")
 
 local Bibliography
 
+local nbsp = luautf8.char(0x00A0)
+local function sanitize (str)
+   local s = str
+      -- TeX special characters:
+      -- Backslash-escaped tilde is a tilde,
+      -- but standalone tilde is a non-breaking space
+      :gsub("(.?)~", function (prev)
+         if prev == "\\" then return "~" end
+         return prev .. nbsp
+      end)
+      -- Other backslash-escaped characters are skipped
+      -- TODO FIXME:
+      -- This ok for \", \& etc. which we want to unescape,
+      -- BUT what should we do with other TeX-like commands?
+      :gsub("\\", "")
+      -- We will wrap the content in <sile> tags so we need to XML-escape
+      -- the input.
+      :gsub("&", "&amp;"):gsub("<", "&lt;"):gsub(">", "&gt;")
+   return s
+end
+
 -- luacheck: push ignore
 -- stylua: ignore start
 ---@diagnostic disable: undefined-global, unused-local, lowercase-global
 local bibtexparser = epnf.define(function (_ENV)
+   local strings = {} -- Local store for @string entries
+
    local identifier = (SILE.parserBits.identifier + S":-")^1
    local balanced = C{ "{" * P" "^0 * C(((1 - S"{}") + V(1))^0) * "}" } / function (...) local t={...}; return t[2] end
-   local doubleq = C( P'"' * C(((1 - S'"\r\n\f\\') + (P'\\' * 1)) ^ 0) * '"' )
+   local quoted = C( P'"' * C(((1 - S'"\r\n\f\\') + (P'\\' * 1)) ^ 0) * '"' ) / function (...) local t={...}; return t[2] end
```

Comment:
To be honest, I refrain touching it too much, because I am not sure the initial rules are actually correct, regarding how the escaping is supposed to be done in the quoted strings. The code is using `P('\\" * 1)`... hinting towards `\"` (which I _think_ is indeed doable), but some manuals say otherwise, using internal braces:
![image](https://github.com/sile-typesetter/sile/assets/18075640/bb040c58-ad01-4a91-86b0-072f8bd6e93f)

And this is a bit tricky, as internal braces (in both cases) are supposed to mean something else too. (An currently we don't support them as intended). Maybe we should have a dedicated issue, and better check what other implementations and manuals recommend here?

Reply:
By the way, the above screen shot is from "Tame the BeaST. The B to X of BibTEX,* version 1.4 (2009).

Reply:
I think you're right the parser probably needs a more aggressive overhaul and I'm happy with just copying the existing function for now. Can you open and issue linking back to this so we come back to it some day? By the way have you run across any ABNF, EBNF, or other formal grammars for BibTeX stuff or are there just a hodge–podge of implementations we're copying that might not even all match each other?

Reply:
c.f. #2075

Reply:
>  Can you open and issue linking back to this so we come back to it some day?

Tracked as #2075 

> By the way have you run across any ABNF, EBNF, or other formal grammars for BibTeX stuff or are there just a hodge–podge of implementations we're copying that might not even all match each other?

For some elements (e.g. `@string`), I checked a few (non-grammar-based) implementation as well as a few ABNF grammars. But the rest is mostly from the biblatex manual and _Tame the Beast._ Problem is that most ABNF specifications I've found are incomplete... So yes, it ends up as a bit of a hodge-podge... (Hence also the "notes" on the format added in the documentation, as it's not clear to me that there's some full complete definition agreed by everyone and formally defined... Biber/biblatex keep adding stuff!)